### PR TITLE
fix(aosh): Adding the skip payment for when it applies for machines

### PR DESCRIPTION
**Pull request by davidkef**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced the transfer process with an optional payment requirement for machine ownership transfers.

- **Enhancements**
  - Updated the machine ownership transfer form to include a hidden input field for handling payment requirements.

- **Bug Fixes**
  - Adjusted application logic for correct transitions between payment and review stages based on the new payment requirement.

- **Documentation**
  - Documented new conditional payment logic and related data fields in the system.

### CODE REVIEW

File: libs/api/domains/work-machines/src/lib/models/machineDetails.ts [hunk 1]
```diff
@@ -31,4 +31,7 @@ export class MachineDetails {
 
   @Field(() => String, { nullable: true })
   supervisorName?: string | null
+
+  @Field(() => Boolean, { nullable: true })
+  paymentRequiredForOwnerChange?: boolean | null
```
Comment (thorkellmani) [inline libs/api/domains/work-machines/src/lib/models/machineDetails.ts:36]:
```suggestion
  @Field({ nullable: true })
  paymentRequiredForOwnerChange?: boolean | null
```

Unnecessary for boolean
image

**Review by thorkellmani — APPROVED** [review]
LGMT

File: libs/application/templates/aosh/transfer-of-machine-ownership/src/utils/index.ts [hunk 2]
```diff
@@ -14,6 +16,13 @@ export { hasReviewerApproved } from './hasReviewerApproved'
 export { getApproveAnswers } from './getApproveAnswers'
 export { getRejecter } from './getRejecter'
 
-export const getChargeItemCodes = (): Array<string> => {
+export const getChargeItemCodes = (
+  application: Application<FormValue>,
+): Array<string> => {
+  const answers = application.answers as TransferOfMachineOwnershipAnswers
+  if (answers.machine.paymentRequiredForOwnerChange === false) {
```
Comment (Ballioli) [inline libs/application/templates/aosh/transfer-of-machine-ownership/src/utils/index.ts:23]: Getur answers.machine.paymentRequiredForOwnerChange verið undefined hérna ?
Ef svo er mun (answers.machine.paymentRequiredForOwnerChange === false) vs (!answers.machine.paymentRequiredForOwnerChange) haga sér mismundandi. Bara henda því út í cosmos-inn ef það hefur áhrif

**Review by Ballioli — APPROVED** [review]

File: libs/application/templates/aosh/transfer-of-machine-ownership/src/utils/index.ts (hunk 2 above)
Comment (davidkef) [reply libs/application/templates/aosh/transfer-of-machine-ownership/src/utils/index.ts:23]: Ætti aldrei að geta verið undefined eða null.
Ætla samt að bæta við null checki í þetta ef það vill svo til

**Pull request merged**